pdf_ocr.go

```diff
@@ -0,0 +1,45 @@
+// +build ocr
+
+package docconv
+
+import (
+	"fmt"
+	"io"
+	"log"
+)
+
+func ConvertPDF(r io.Reader) (string, map[string]string, error) {
+	log.Println("checking if pdf has image.... ")
+	f, err := NewLocalFile(r, "/tmp", "sajari-convert-")
+	if err != nil {
+		return "", nil, fmt.Errorf("error creating local file: %v", err)
+	}
+	defer f.Done()
+
+	// Verify if pdf has images or is pdf only-text
+	log.Println("checking if pdf has image... ")
+	if PDFHasImage(f.Name()) {
+		log.Println("pdf does have image")
+		bodyResult, imageConvertErr := ConvertImagePDF(f.Name())
+		if imageConvertErr != nil {
+			return "", nil, imageConvertErr
+		}
+		if bodyResult.err != nil {
+			return "", nil, bodyResult.err
+		}
+
+		return bodyResult.body, nil, nil
```

Comment:
If a PDF has an image, the text is totally ignored. This will fail for mixed PDFs. Needs to handle both or explicitly prevent